### PR TITLE
Cleaner documentation

### PR DESCRIPTION
**Pull request by wakakusa**
New function documentation has been created and function descriptions have been removed from README.md.

Issue the key by executing the following command:.
```
using DocumenterTools,LightGBM

DocumenterTools.genkeys(;user=IQVIA-ML, repo="git@github.com:IQVIA-ML/LightGBM.jl.git")
```
Please add the public key to https://github.com/IQVIA-ML/LightGBM.jl.git/settings/keys with read/write access and add a secure environment variable named 'DOCUMENTER_KEY' to https://github.com/IQVIA-ML/LightGBM.jl.git/settings/secrets

If set, document creation will be automatically executed and published every time commit.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,6 +3,8 @@ This package was originally authored by [Allardvm](https://github.com/Allardvm)
 LightGBM.jl
 ![CI](https://github.com/IQVIA-ML/LightGBM.jl/workflows/CI/badge.svg)
 [![License](http://img.shields.io/badge/license-MIT-brightgreen.svg?style=flat)](LICENSE.md)
+[![Stable](https://img.shields.io/badge/docs-stable-blue.svg)](https://IQVIA-ML.github.io/LightGBM.jl/stable)
+[![Dev](https://img.shields.io/badge/docs-dev-blue.svg)](https://IQVIA-ML.github.io/LightGBM.jl/dev)
```
Comment (yalwan-iqvia) [inline README.md:7]: Whats the difference between dev and stable?

File: docs/src/index.md [hunk 2]
```diff
@@ -1,8 +1,131 @@
...
+The MLJ interface models are
+```julia
+LightGBM.MLJInterface.LGBMClassifier
+LightGBM.MLJInterface.LGBMRegressor
 ```
+
+And these have the same interface parameters as the [estimators](#estimators)
+
+The interface models are generally passed to `MLJBase.fit` or `MLJBase.machine`
+and integrated as part of a larger MLJ pipeline. [An example is provided](https://alan-turing-institute.github.io/MLJTutorials/end-to-end/boston-lgbm/)
```
Comment (yalwan-iqvia) [inline docs/src/index.md:131]: The link is broken it needs updating to this: (that's not your fault)

https://alan-turing-institute.github.io/DataScienceTutorials.jl/end-to-end/boston-lgbm/

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -43,3 +43,20 @@ jobs:
         continue-on-error: false
         env:
           LIGHTGBM_EXAMPLES_PATH: LightGBM-2.3.1
+ docs:
```
Comment (yalwan-iqvia) [inline .github/workflows/main.yml:46]: Do we want to restrict this to running on master?

**Comment by yalwan-iqvia** [comment]
Hi @wakakusa ! 

Thanks for taking the time to contribute this. I will make a few comments here and there but it should be ok. Also it looks like at the moment that the CI has been broken. We will need to resolve this and then hopefully we can get this great contribution merged!

**Comment by yalwan-iqvia** [comment]
I think the CI will need a bit of work on the basis of this one:

https://docs.github.com/en/actions/reference

I'll go through it and make some suggestions for you

File: README.md (hunk 1 above)
Comment (wakakusa) [reply README.md:7]: The development version is the document that created the document from the latest committed code.
A stable version is a document created from the latest release of code.

File: .github/workflows/main.yml (hunk 3 above)
Comment (wakakusa) [reply .github/workflows/main.yml:46]: I think only execution on the master is enough.

File: docs/src/index.md (hunk 2 above)
Comment (wakakusa) [reply docs/src/index.md]: The link was broken and was updated as follows (https://alan-turing-institute.github.io/DataScienceTutorials.jl/end-to-end/boston-lgbm/)

File: README.md (hunk 1 above)
Comment (yalwan-iqvia) [reply README.md:7]: Sorry I mean to clarify -- how will the system know whats dev and whats stable? Will we have to do something periodically manually to make the right things appear at dev/stable?

Comment (wakakusa) [reply README.md:7]: The system automatically determines if the document is stable based on the release tag. So you don't have to do anything manually on a regular basis.

File: .github/workflows/main.yml (hunk 3 above)
Comment (yalwan-iqvia) [inline .github/workflows/main.yml:46]: So this one is the reason why the CI isn't working -- the indentation for `docs:` needs to be 2, not 1, then it should trigger the CI correctly.

I think we need to add something like this:

```
  docs:
    on:
      push:
        branches: master
        tags: *
```

**Comment by yalwan-iqvia** [comment]
I've set this up against my own fork of your fork, trying to get the documentation to show -- unfortunately I dont have a working github pages site yet -- could you possibly take a look at https://github.com/yalwan-iqvia/LightGBM.jl-1

and see what i have done wrong? I set the the documenter key with credentials for that repo. Do the logs show anything when it published successfully? all the actions have just shown something like this:
```
Run julia --project=docs docs/make.jl
[ Info: SetupBuildDirectory: setting up build directory.
[ Info: Doctest: running doctests.
[ Info: ExpandTemplates: expanding markdown templates.
[ Info: CrossReferences: building cross-references.
[ Info: CheckDocument: running document checks.
[ Info: Populate: populating indices.
[ Info: RenderDocument: rendering document.
[ Info: HTMLWriter: rendering HTML pages.
````

p.s. this is after I fixed the syntax to incrwase to indent of the docs: section by 1 space

**Comment by yalwan-iqvia** [comment]
Ok, after lots and lots of fiddling in my own private repo, I managed to grok this finally.

I think you just make those changes to the yml we can handle the rest, however if you'd like to contribute more I can suggest other improvements at the readme and how we link it all together (e.g. the exports section in the readme is no longer relevant)

Let me know how you'd like to continue

**Comment by wakakusa** [comment]
I would like to continue to contribute, so could you tell me other improvements of readme and how to link them?

File: README.md [hunk 4]
```diff
@@ -118,236 +120,6 @@ exact matches with those from [LightGBM.](https://lightgbm.readthedocs.io/en/lat
 Not all of these are necessarily supported but see the guide for detailed explanations of what these
```
Comment (yalwan-iqvia) [inline README.md:120]: This whole section can be removed and its corresponding part in the table of contents.

File: docs/src/index.md [hunk 5]
```diff
@@ -1,8 +1,131 @@
...
+
+# Save and load the fitted model.
+filename = pwd() * "/finished.model"
+savemodel(estimator, filename)
+loadmodel(estimator, filename)
 ```
 
-```@autodocs
-Modules = [LightGBM]
+# Exports
```
Comment (yalwan-iqvia) [inline docs/src/index.md:110]: You could probably rename this to Parameters and the same in the TOC

File: docs/src/function.md [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# LightGBM.jl
```
Comment (yalwan-iqvia) [inline docs/src/function.md:1]: This file should probably be renamed to "Functions" or "API"  based on how it appears in the generated pages site

Also does it make sense for the header to be LightGBM.jl or something else?

**Comment by yalwan-iqvia** [comment]
> I would like to continue to contribute, so could you tell me other improvements of readme and how to link them?

I have made a few more comments, if you could address those and fix the CI with the part I commented on previously then hopefully we can be closer to merging. Thanks for your efforts so far!

File: docs/src/function.md (hunk 6 above)
Comment (wakakusa) [reply docs/src/function.md:1]: Rename the file "Functions".
Also modify the header to a "Functions".

File: README.md (hunk 4 above)
Comment (wakakusa) [reply README.md:120]: Remove the "Export" section and its table of contents.

File: docs/src/index.md (hunk 5 above)
Comment (wakakusa) [reply docs/src/index.md]: Rename the section "Export" to "Parameters". Modify the table of contents in the same way

**Comment by wakakusa** [comment]
Reflected the contents of the review

**Comment by yalwan-iqvia** [comment]
Hi, thanks for the updates. Theres one major thing missing:
https://github.com/IQVIA-ML/LightGBM.jl/pull/67/files#diff-90829e76e906f1c73140c7ded7e1b268R46

Please see above link. The indentation here is incorrect, it needs to be 2 spaces rather than 1, and then I think the CI will run

**Comment by wakakusa** [comment]
I'm sorry. Corrected the indentation of main.yml to two spaces. The functions.md file was not uploaded, so we have corrected it accordingly.

**Comment by yalwan-iqvia** [comment]
Thanks wakakkusa, I will merge this, if you think it needs something else, you can open another PR, but I'm satisfied with the current state. (obviously the objective is to continuously improve)

**Pull request merged**